url835a90ed59_aschi_regularLanguages_tgz-pJ8-ch_zhaw_regularLanguages_evolution_runners_EvolvingLocalDivisable5LanguageJ8/src/ch/zhaw/regularLanguages/evolution/EAWithEvolvingLocalProblemSets.java

```diff
@@ -0,0 +1,122 @@
+package ch.zhaw.regularLanguages.evolution;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import ch.zhaw.regularLanguages.evolution.candidates.EvolutionCandidate;
+import ch.zhaw.regularLanguages.evolution.candidates.EvolutionCandidateWithProblemSet;
+import ch.zhaw.regularLanguages.evolution.problems.ProblemSet;
+import ch.zhaw.regularLanguages.helpers.PublicCloneable;
+
+public class EAWithEvolvingLocalProblemSets<E extends EvolutionCandidateWithProblemSet, R> implements EvolutionaryAlgorithm<E>{
+	private List<E> candidates;
+	
+	private E winner;
+	
+	private R reference;
+	
+	private int maxFitness;
+	private int maxC;
+	
+	
+	public EAWithEvolvingLocalProblemSets(List<E> candidates, R reference, int noProblems) {
+		this.candidates = candidates;
+		this.reference = reference;
+		this.maxFitness = noProblems;
+	}
+	
+	public E getWinner(){
+		return winner;
+	}
+	
+	public int getMaxC(){
+		return maxC;
+	}
+		
+	public List<E> getCandidates(){
+		return candidates;
+	}
+
+	@Override
+	public long startEvolution(long cycleLimit) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.